app/jobs/user_banned_password_check_job.rb

```diff
@@ -1,13 +1,20 @@
 class UserBannedPasswordCheckJob < ApplicationJob
   queue_as :user_password_check
 
-  def perform(user_id)
+  BATCH_SIZE = 10
+
+  def perform(user_id, offset = 0)
     user = User.find(user_id)
+    denylist = BannedPassword.limit(BATCH_SIZE).offset(offset).pluck(:password)
 
-    has_banned_password = BannedPassword.pluck(:password).any? do |password|
+    has_banned_password = denylist.any? do |password| # pragma: allowlist secret
       user.valid_password?(password)
     end
 
-    user.update!(banned_password_match: has_banned_password)
+    if denylist.length < BATCH_SIZE || has_banned_password
```

Comment:
If the full `BannedPassword` list is a multiple of `BATCH_SIZE`, I think the else will get triggered, even if the list has been fully processed.  The next job should rectify this though, since `denylist` will be zero in length, so I think this is ok (if we're happy to have extra jobs queued in the unlikely event the full list is divisible by 10).